pkg/prom/query.go

```diff
@@ -172,18 +184,24 @@ func runQuery(query string, ctx *Context, resCh QueryResultsChan, profileLabel s
 }
 
 // RawQuery is a direct query to the prometheus client and returns the body of the response
-func (ctx *Context) RawQuery(query string) ([]byte, error) {
+func (ctx *Context) RawQuery(query string, t time.Time) ([]byte, error) {
 	u := ctx.Client.URL(epQuery, nil)
 	q := u.Query()
 	q.Set("query", query)
 
-	// for non-range queries, we set the timestamp for the query to time-offset
-	// this is a special use case that's typically only used when our primary
-	// prom db has delayed insertion (thanos, cortex, etc...)
-	if promQueryOffset != 0 && ctx.name != AllocationContextName {
-		q.Set("time", time.Now().Add(-promQueryOffset).UTC().Format(time.RFC3339))
+	if !t.IsZero() {
+		// TODO remove log
+		log.Infof("[Prom] time=%s query=%s", strconv.FormatInt(t.Unix(), 10), query)
+		q.Set("time", strconv.FormatInt(t.Unix(), 10))
 	} else {
-		q.Set("time", time.Now().UTC().Format(time.RFC3339))
+		// for non-range queries, we set the timestamp for the query to time-offset
+		// this is a special use case that's typically only used when our primary
+		// prom db has delayed insertion (thanos, cortex, etc...)
+		if promQueryOffset != 0 && ctx.name != AllocationContextName {
+			q.Set("time", time.Now().Add(-promQueryOffset).UTC().Format(time.RFC3339))
+		} else {
+			q.Set("time", time.Now().UTC().Format(time.RFC3339))
+		}
```

Comment:
Likewise, here, want to be very careful @mbolt35

Reply:
Yeah, I love these changes, but I am bit concerned that this will break the intention behind the hosted hacks (which I'm still not certain why allocation breaks). 

Nothing outside of very specific circumstances should we actually set the prom query offset, so while I would love to discontinue the offset hackery, I think the only way to keep it consistent is the add any t time.Time to the promQueryOffset.

We can discuss this more if you have more specific concerns, but if you can imagine every prod deployment of our product having promQueryOffset=0, that'd be great for now :-)